cypress/support/commands.js

```diff
@@ -23,3 +24,34 @@
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+Cypress.Commands.add('login', (user = {
+  email: Cypress.env('email'),
+  password: Cypress.env('password')
+}) => {
+  cy.visit('/login');
+  cy.get('#email').should('be.visible').type(user.email)
+  cy.get('#password').should('be.visible').type(user.password)
+  cy.get('button[type="submit"]').should('be.visible').click()
+  cy.url().should('eq', 'https://notes-serverless-app.com/')  
+})
+
+Cypress.Commands.add('createsANote', data => {
+  cy.contains('Create a new note').should('be.visible').click()
+  cy.get('#content').should('be.visible').type(data.content)
+  cy.contains('Create').should('be.visible').click()
+  cy.get('.list-group').should('contain', data.content)
+})
+
+Cypress.Commands.add('editsANote', data => {
+  cy.get('.list-group').contains(data.content).should('be.visible').click()
+  cy.get('#content').should('be.visible').clear().type(data.newContent)
+  cy.contains('Save').should('be.visible').click()
+  cy.get('.list-group').should('contain', data.newContent)
+  cy.get(`.list-group:contains(${data.newContent})`).should('be.visible')
+})
+
+Cypress.Commands.add('deletesANote', data => {
+  cy.get('.list-group').contains(data.newContent).click()
+  cy.contains('Delete').click()
+  cy.get(`.list-group:contains(${data.newContent})`).should('not.exist')
```

Comment:
Ideally, we should wait for the redirect to happen before running the assertion. Also, it would be better if the assertion was on the test itself, not the custom command.

Reply:
Done! I moved all of the assertions from the custom commands to the test body.